### ISSUE TITLE
Fetching a token displays confusing alert message to users

### ISSUE THREAD

**Issue by yepthatsjason** [aws-amplify/aws-sdk-ios#2429]
**Describe the bug**
Our app needs to be able to send the JWT ID token to our AWS java backend service so it can validate user API calls. I've noticed this annoying message that shows up whenever the access token needs to get updated (which is extremely frequent), and is displayed even though the user state is already "signedIn". It's not reasonable to display this message to our app users, and furthermore people probably have no idea what "amazoncognito.com" is and why we're sharing information with them. Screenshot below.

How can I get an access token to authenticate with my backend service WITHOUT showing this alert message every time it's updated?

**To Reproduce**
1. Login a user
2. Quit the app
3. Re-open the app after 1hr, user state moves to signedIn automatically
4. Try and get an access token, which triggers the alert message to appear.

```
AWSMobileClient.default().getTokens { (tokens, error) in
      if let tokens = tokens {
        print("id token: \(tokens.idToken?.tokenString)")
        print("access token: \(tokens.accessToken?.tokenString)")
      }
    }
```

**Which AWS service(s) are affected?**
AWS Cognito, AWS Amplify, AWS RDS, AWS Elastic Beanstalk

**Expected behavior**
We should be able to get a valid access token after calling `AWSMobileClient.default().getTokens` and it shouldn't display a UI alert message.

**Screenshots**
Here's a demo app showing the problem. The `logout` button shows we're currently signed in, and when I call `AWSMobileClient.default().getTokens` it shows this alarming UI every time the token needs to be updated (which seems like every hour).

![Simulator Screen Shot - iPhone 11 Pro - 2020-04-13 at 10 42 15](https://user-images.githubusercontent.com/246399/79144830-7a509080-7d74-11ea-8833-aa7a4d6b59bc.png)

**Environment(please complete the following information):**
 - SDK Version: AWSMobileClient 2.13.1
 - Dependency Manager: Cocoapods
 - Swift Version : 5

**Device Information (please complete the following information):**
 - Device: iOS Simulator
 - iOS Version: iOS 13

**Additional context**
This is unfortunately a blocker for us to use the authentication built into AWSMobileClient to use Cognito. Any help would be greatly appreciated.

**Comment by yepthatsjason** [comment]
Additionally, I noticed that after I click this message, it takes me to the built-in web UI login. Which then asks me to login, and after I did.. the it returned and the `getTokens(..)` completed. 

Hopefully I'm just doing something wrong here, but it seems like Cognito is making the user perform a full login every time the access token expires, even though their client state is set to "signedIn". I was able to verify that the user was signed in before calling `getTokens(..)` which made me assume that that I would be able to properly pull the updated token.

```
    AWSMobileClient.default().initialize { (userState, error) in
      if let userState = userState, userState == .signedIn {
          print("User is logged in, but getTokens shows the login UI?!?!")

          AWSMobileClient.default().getTokens { (tokens, error) in
            if let tokens = tokens {
              print("id token: \(tokens.idToken?.tokenString)")
              print("access token: \(tokens.accessToken?.tokenString)")
            }
          }
        } else if let error = error {
            print("error: \(error.localizedDescription)")
        }
    }
```

**Comment by yepthatsjason** [comment]
It seems like the only reason this alert UI would appear is if the login API wasn't properly maintain access to the Google Sign-In refresh token? Is there possibly an issue in the login flow for native that's not storing the token as expected?

I'm logging in like this into my AWS Cognito User Pool:
```
    let hostedUIOptions = HostedUIOptions(scopes: ["openid", "email", "profile"], identityProvider: IdentityProvider.google.rawValue)

    AWSMobileClient.default().showSignIn(navigationController: self.navigationController!, hostedUIOptions: hostedUIOptions) { (signInState, error) in
        if let signInState = signInState {
            print("Sign in flow completed: \(signInState)")
        } else if let error = error {
            print("error logging in: \(error.localizedDescription)")
        }
    }
```

**Comment by yepthatsjason** [comment]
I noticed today that the login session did in fact persist over the night as expected. I'm wondering if maybe there's just a weird simulator issue happening here with the web views credential cash or something, because when I changed to a different iOS simulator that was also previously authenticated, Cognito thought I was signed-in, but it prompted asking to re-open the web view to authenticate and the web view didn't have my existing google login information. This makes me think that the web cache got cleared somehow, which hopefully wouldn't be happening on a real iOS device. I'm going to close this issue as it might just be a simulator problem or something in my workflow causing it.
